apply --type-check is broken

This used to work at some point.

Now I get:

```
> chisel apply --type-check
Error: could not bundle endpoints with esbuild (using node-style modules)

Caused by:
    endpoints/hello.ts:7:21 - error TS2339: Property 'crud' does not exist on type 'typeof User'.

    7 export default User.crud()
                          ~~~~

    models/User.ts:1:30 - error TS2307: Cannot find module '@chiselstrike/api' or its corresponding type declarations.

    1 import { ChiselEntity } from "@chiselstrike/api"
                                   ~~~~~~~~~~~~~~~~~~~
```

Some bad option on tsconfig?

my bad, I had a local issue.

The real error is this:
```
Caused by:
    error TS5023: Unknown compiler option '--checkJs9nwqbm00000gn/T/.tmpxxlBGr/1.ts'.
    error TS5023: Unknown compiler option '--bundle'.
    error TS5023: Unknown compiler option '--color=true'.
    error TS5023: Unknown compiler option '--target=esnext'.
    error TS5023: Unknown compiler option '--external:@chiselstrike'.
    error TS5023: Unknown compiler option '--format=esm'.
    error TS5023: Unknown compiler option '--tree-shaking=true'.
    error TS5023: Unknown compiler option '--tsconfig=./tsconfig.json'.
    error TS5023: Unknown compiler option '--platform=node'.
    error TS5023: Unknown compiler option '--outdir=/var/folders/7z/zhbwn_b15dj2kwm3n9nwqbm00000gn/T/.tmplcyEwT'.
```

I think we can't really (unfortunately) call npx in parallell for both tsc and esbuild. We had this issue before. it seems like somehow tsc ends up with the esbuild options

--- a/cli/src/cmd/apply/node.rs
+++ b/cli/src/cmd/apply/node.rs
@@ -30,6 +30,16 @@ pub(crate) async fn apply(
         )),
         TypeChecking::No => None,
     };
+    // ideally we would call this in parallel with the bundle, but npx doesn't like this very much
+    // See #1642
+    if let Some(tsc) = tsc {
+        let tsc_res = tsc.await.unwrap()?;
+        if !tsc_res.status.success() {
+            let out = String::from_utf8(tsc_res.stdout).expect("command output not utf-8");
+            let err = String::from_utf8(tsc_res.stderr).expect("command output not utf-8");
+            anyhow::bail!("{}\n{}", out, err);
+        }
+    }
 
     let cwd = env::current_dir()?;
     let gen_dir = cwd.join(".gen");
@@ -148,14 +158,6 @@ pub(crate) async fn apply(
         }
     }
 
-    if let Some(tsc) = tsc {
-        let tsc_res = tsc.await.unwrap()?;
-        if !tsc_res.status.success() {
-            let out = String::from_utf8(tsc_res.stdout).expect("command output not utf-8");
-            let err = String::from_utf8(tsc_res.stderr).expect("command output not utf-8");
-            anyhow::bail!("{}\n{}", out, err);
-        }
-    }
     Ok((sources, index_candidates))
 }
 

